### PR TITLE
Use django rest framework

### PR DESCRIPTION
**Pull request by arturo-seijas**
Applicable spec: N/A

### Overview

Install the Django REST framework and leverage its authentication features

### Rationale

Django REST framework will be needed to support JWT auth

### Module Changes

settings and views

### Checklist

- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

File: app/settings.py [hunk 1]
```diff
@@ -75,14 +75,20 @@
...
 DATABASES = {
     "default": {
-        "ENGINE": environ.get("DB_ENGINE"),
-        "NAME": environ.get("DB_NAME"),
-        "USER": environ.get("DB_USER"),
-        "PASSWORD": environ.get("DB_PASSWORD"),
-        "HOST": environ.get("DB_HOST"),
-        "PORT": environ.get("DB_PORT"),
+        "ENGINE": "django.db.backends.sqlite3",
+        "NAME": BASE_DIR / "db.sqlite3",
```
Comment (mthaddon) [inline app/settings.py:91]: Won't this cause problems when we come to the point where we want to build/deploy a charm for this?

Comment (arturo-seijas) [reply app/settings.py]: That change shouldn't be there. My bad

**Review by mthaddon — DISMISSED** [review]
LGTM, thx

**Review by mthaddon — APPROVED** [review]
LGTM, thx

**Review by nrobinaubertin — APPROVED** [review]

**Pull request merged**